xtask/src/ci/target_matrix.rs

```diff
@@ -0,0 +1,111 @@
+use clap::builder::BoolishValueParser;
+use clap::Parser;
+use serde::Serialize;
+
+use crate::util::{get_matrix, gha_output, gha_print, CiTarget, ImageTarget};
+
+pub(crate) fn run(message: String, author: String) -> Result<(), color_eyre::Report> {
+    let mut matrix: Vec<CiTarget> = get_matrix().clone();
+    if author == "bors[bot]" && message.starts_with("Try #") {
+        if let Some((_, args)) = message.split_once(": ") {
+            let app = TargetMatrixArgs::parse_from(args.split(' '));
+            if app.target.is_empty() {
+            } else {
+                matrix.retain(|m| {
```

Comment:
Does this support wildcards? It would be nice to expand globs, if possible.

Reply:
ah no, I forgot about that. I'm not super enthustiastic about adding regex to the deps here, but with caching I guess it's fine, our xtask is already pretty heavy

Reply:
and for globs specifically, it could be possible to implement with some algorithm I'm not aware of easily I suppose

Reply:
I don't want regex itself, just glob support which I think we can do. I wonder if we can have a lightweight matcher for just the expansion of `*` characters.

Reply:
supporting trailing `*` would be easy

Reply:
As mentioned in Matrix, it looks like [wildmatch](https://crates.io/crates/wildmatch/2.1.1/dependencies) is a lightweight and good fit for this. We might also want some very simple tests for what the matrix matcher produces (check if some targets we'll always have are retained).